pkg/internal/limitrange/transformer.go

```diff
@@ -45,98 +46,78 @@ func NewTransformer(ctx context.Context, namespace string, lister corev1listers.
 
 		// The assumption here is that the min, max, default, ratio have already been
 		// computed if there is multiple LimitRange to satisfy the most (if we can).
-		// Count the number of containers (that we know) in the Pod.
+		// Count the number of "step" containers in the Pod.
 		// This should help us find the smallest request to apply to containers
-		nbContainers := len(p.Spec.Containers)
-		// FIXME(#4230) maxLimitRequestRatio to support later
-		defaultLimits := getDefaultLimits(limitRange)
-		defaultInitRequests := getDefaultInitContainerRequest(limitRange)
-		for i := range p.Spec.InitContainers {
-			// We are trying to set the smallest requests possible
-			if p.Spec.InitContainers[i].Resources.Requests == nil {
-				p.Spec.InitContainers[i].Resources.Requests = defaultInitRequests
-			} else {
-				for _, name := range resourceNames {
-					setRequestsOrLimits(name, p.Spec.InitContainers[i].Resources.Requests, defaultInitRequests)
-				}
-			}
-			// We are trying to set the highest limits possible
-			if p.Spec.InitContainers[i].Resources.Limits == nil {
-				p.Spec.InitContainers[i].Resources.Limits = defaultLimits
-			} else {
-				for _, name := range resourceNames {
-					setRequestsOrLimits(name, p.Spec.InitContainers[i].Resources.Limits, defaultLimits)
-				}
+		stepContainers := []*corev1.Container{}
+		for i, c := range p.Spec.Containers {
+			if strings.HasPrefix(c.Name, "step-") {
+				stepContainers = append(stepContainers, &p.Spec.Containers[i])
 			}
 		}
+		// No step containers, bail early we don't have anything to transform.
```

Comment:
This will change existing behavior for init containers (currently the defaultrequest and default is applied to init containers, and now it will not be)

Reply:
This does not change behavior as this is what the LimitRanger does already. We are in effect duplicating the work that this does. We only need to do this work for the Step containers as we are doing request splitting. For limits and for managing the LimitRange's for other containers in the Pod we should just let the LimitRanger do its thing.

Reply:
ah right, I think this will work since init containers won't have any resource requirements set.

Reply:
There is one difference that is worth discussing. With the existing Tekton transformer if a container has set a limit that is "higher" than the max limit the transformer will overwrite with the max limit value. For the same situation the LimitRanger will throw an error on creation -- see https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/cpu-constraint-namespace/#attempt-to-create-a-pod-that-exceeds-the-maximum-cpu-constraint

This is a change in functionality however I think it's unlikely anyone was relying on this silent downgrading. If we can avoid it we probably should stick with the existing LimitRange semantics but I wanted to call it out.

Reply:
Yeah, this is what I was attempting to get at in the other comment thread :) We do document this behavior and it's hard to know if anyone is relying on it or not. There are multiple options for what the "correct" behavior should be, so I'd prefer to keep the scope of this PR to just setting the correct defaults for sidecars, and if we want to change the behavior of limits, let's discuss that separately.

Reply:
ok... let me roll back time a bit here.